optimum/habana/transformers/models/starcoder2/modeling_starcoder2.py

```diff
@@ -307,7 +308,8 @@ def pre_attn_forward(
 
             if q_len == 1:
                 # next token
-                with ht.sdp_kernel(enable_recompute=False):
+                use_recompute = True if os.getenv("QUANT_CONFIG", "") else False
```

Comment:
```suggestion
                use_recompute = bool(os.getenv("QUANT_CONFIG"))
```

personal suggestion, less code!